src/techs/techs.controller.ts

```diff
@@ -228,6 +232,133 @@ export class TechsController {
         return this.techsService.deleteTeamTech(req, teamTechItemId);
     }
 
+    @ApiOperation({
+        summary:
+            "[Permission: own_team] Adds a new tech stack category to the team.",
+        description: "Requires login",
+    })
+    @ApiResponse({
+        status: HttpStatus.CREATED,
+        description: "Successfully added a new tech stack category",
+        type: TechCategoryResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.CONFLICT,
+        description: "Tech stack category already exist for the team",
+        type: ConflictErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.BAD_REQUEST,
+        description: "Invalid userId",
+        type: BadRequestErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.UNAUTHORIZED,
+        description: "Unauthorized access - user is not logged in",
+        type: UnauthorizedErrorResponse,
+    })
+    @Post("teams/:teamId/techStackCategory")
+    addNewTechStackCategory(
+        @Request() req: CustomRequest,
+        @Param("teamId", ParseIntPipe) teamTId: number,
+        @Body(ValidationPipe, VoyageTeamMemberValidationPipe)
+        createTechStackCategoryDto: CreateTechStackCategoryDto,
+    ) {
+        return this.techsService.addNewTechStackCategory(
+            req,
+            createTechStackCategoryDto,
+        );
+    }
+
+    @ApiOperation({
+        summary:
+            "[Permission: own_team] Updates an existing tech stack category in the team",
+        description: "Requires login",
+    })
+    @ApiResponse({
+        status: HttpStatus.OK,
+        description: "Successfully updated a tech stack category",
+        type: TechCategoryResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.UNAUTHORIZED,
+        description: "unauthorized access - user is not logged in",
+        type: UnauthorizedErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.FORBIDDEN,
+        description: "forbidden - user does not have the required permission",
+        type: ForbiddenErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.BAD_REQUEST,
+        description: "Bad Request - tech stack category couldn't be updated",
+        type: BadRequestErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.CONFLICT,
+        description: "Tech stack category already exists for the team",
+        type: ConflictErrorResponse,
+    })
+    @ApiResponse({
+        status: HttpStatus.NOT_FOUND,
+        description: "Invalid tech stack category id",
+        type: NotFoundErrorResponse,
+    })
+    @Patch("teams/techStackCategory")
```

Comment:
I think we should have the techstackCategoryId in the uri instead of the body as it identifies the resource